docs/apis/streaming/windows.md

```diff
@@ -24,1023 +24,608 @@ specific language governing permissions and limitations
 under the License.
 -->
 
+Flink uses a concept called *windows* to divide a (potentially) infinite `DataStream` into finite
+slices based on the timestamps of elements or other criteria. This division is required when working
+with infinite streams of data and performing transformations that aggregate elements.
+
+<span class="label label-info">Info</span> We will mostly talk about *keyed windowing* here, i.e.
+windows that are applied on a `KeyedStream`. Keyed windows have the advantage that elements are
+subdivided based on both window and key before being given to
+a user function. The work can thus be distributed across the cluster
+because the elements for different keys can be processed independently. If you absolutely have to,
+you can check out [non-keyed windowing](#non-keyed-windowing) where we describe how non-keyed
+windows work.
+
 * This will be replaced by the TOC
 {:toc}
 
-## Windows on Keyed Data Streams
-
-Flink offers a variety of methods for defining windows on a `KeyedStream`. All of these group elements *per key*,
-i.e., each window will contain elements with the same key value.
+## Basics
 
-### Basic Window Constructs
+For a windowed transformations you must at least specify a *key*
```

Comment:
Are we using the oxford comma in our doc, I didn't think so? 😄 

But fixing it and the `transformations`.